jdaviz/app.py

```diff
@@ -733,6 +737,36 @@ def _link_new_data(self, reference_data=None, data_to_be_linked=None):
             dc.add_link(links)
             return
 
+        elif self.config == 'rampviz':
+            if linked_data.ndim == 1:
+                # load a 1D ramp profile:
+                ref_group_component = dc[0].components[-2]
+                ref_flux_component = dc[0].components[-1]
+                linked_group_component = dc[-1].components[0]
+                linked_flux_component = dc[-1].components[1]
+
+                links = [
+                    LinkSame(ref_group_component, linked_group_component),
+                    LinkSame(ref_flux_component, linked_flux_component)
+                ]
+
+                dc.add_link(links)
+                return
+            elif linked_data.ndim == 2:
+                # load a several 1D ramp profiles from a 2D NDDataArray:
+                ref_group_component = dc[0].components[-2]
+                ref_flux_component = dc[0].components[-1]
+                linked_group_component = dc[-1].components[1]
+                linked_flux_component = dc[-1].components[2]
+
+                links = [
+                    LinkSame(ref_group_component, linked_group_component),
+                    LinkSame(ref_flux_component, linked_flux_component)
+                ]
+
+                dc.add_link(links)
+                return
```

Comment:
This case may or may not actually be useful if we want to enable loading more than one 1D ramp profile in the integration viewer from a 2D NDDataArray. I'm asking Tom+Derek for advice on whether or not that's plausible to implement.